examples/pre_put_check/main.go

```diff
@@ -55,16 +55,24 @@ func main() {
 		}
 
 		fmt.Println("Item size:", size, "bytes")
+		// I wonder what the performance cost of this Is
+		// I think thsi really make sens to add to hologram if the performance cost is low
+		//  could do cool things as catching early or dynamically split objects.
```

Comment:
Good question, the object needs to marshaled anyway before `Put` to dynamo, which I would bet is the biggest performance cost? 

Traversing the marshaled type recursively and summing their size (`+=len(attribute/value)`) would be proportional to the number of items n `O(n)`, and not proportional to their size. Maybe I can do some benchmarking...